upgrade-job/src/helm/client.rs

```diff
@@ -0,0 +1,154 @@
+
+        info!(%command, ?args, "Helm list command");
+
+        let output = Command::new(command)
+            .args(args.clone())
+            .output()
+            .map_err(|e| HelmCommand {
+                source: e,
+                command: command.to_string(),
+                args,
+            })?;
+
+        let stdout_str =
+            str::from_utf8(output.stdout.as_slice()).map_err(|e| U8VectorToString { source: e })?;
+        info!(stdout=%stdout_str, "Helm list command standard output");
+        if !output.stderr.is_empty() {
+            let stderr_str = str::from_utf8(output.stderr.as_slice())
+                .map_err(|e| U8VectorToString { source: e })?;
+            info!(stderr=%stderr_str, "Helm list command standard error");
+        }
+
+        serde_yaml::from_slice(output.stdout.as_slice()).map_err(|e| YamlParseFromSlice {
+            source: e,
+            input_yaml: stdout_str.to_string(),
+        })
+    }
+
+    pub(crate) fn upgrade(
+        &self,
+        release_name: String,
+        chart_dir: String,
+        maybe_extra_args: Option<Vec<String>>,
+    ) -> Result<()> {
+        let command: &str = "helm";
+        let mut args: Vec<String> = vec![
+            "upgrade".to_string(),
+            release_name,
+            chart_dir,
+            "-n".to_string(),
+            self.namespace.clone(),
+        ];
+
+        if let Some(extra_args) = maybe_extra_args {
+            for arg in extra_args.into_iter() {
+                args.push(arg);
+            }
+        }
+
+        info!(%command, ?args, "Helm upgrade command");
+
+        let output = Command::new(command)
+            .args(args.clone())
+            .output()
+            .map_err(|e| HelmCommand {
+                source: e,
+                command: command.to_string(),
+                args,
+            })?;
+
+        let stdout_str =
+            str::from_utf8(output.stdout.as_slice()).map_err(|e| U8VectorToString { source: e })?;
+        info!(stdout=%stdout_str, "Helm upgrade command standard output");
+        if !output.stderr.is_empty() {
+            let stderr_str = str::from_utf8(output.stderr.as_slice())
+                .map_err(|e| U8VectorToString { source: e })?;
+            info!(stderr=%stderr_str, "Helm upgrade command standard error");
+        }
+
+        info!("Helm upgrade successful!");
+
+        Ok(())
+    }
+
+    pub(crate) fn release_info(&self, release_name: String) -> Result<HelmReleaseElement> {
+        let release_list = self.list(None)?;
+
+        for release in release_list.into_iter() {
+            if release.name().eq(&release_name) {
+                return Ok(release);
+            }
+        }
+
+        // The code reaching this line means that the release is not there, even though we might
+        // have seen that it exists some while back when validating the input Helm release
+        // name in the input Namespace.
+        panic!(
```

Comment:
why not return an error to the upper "levels"?

Reply:
Got rid of it!